sqlrunner/main.py

```diff
@@ -34,6 +36,26 @@ def main(args):
     write_results(results, args["output"])
 
 
+def generate_column_headers(sql_query):
+    tsql_parser = TSQL()
+
+    # filter out any empty expressions returned by parser
+    parsed = [q for q in tsql_parser.parse(sql_query) if q]
+
+    # expand out "*" to full column names if referencing another object
+    # in query with explicit column names
+    columns = qualify_columns(parsed[-1], schema=None, infer_schema=True).named_selects
+
+    # if "*" not referencing an object with explicit column names
+    # we don't know what the headers should be
+    if "*" in columns:
+        raise RuntimeError(
+            "Headers can only be generated for queries with explicit column names"
+        )
+
+    yield {c: None for c in columns}
```

Comment:
I can see why you've chosen to yield a dict, here. However, we end up with a function that doesn't quite do what we expect it to do. Why not return `columns` and then create a `dict` with the correct keys but `None` values in `main`? Doing so would make the corresponding test much easier to reason about.